src/includes/class-alma-wc-product-handler.php

```diff
@@ -83,4 +81,29 @@ public function inject_payment_plan( $the_product = false ) {
 			$amount_query_selector
 		);
 	}
+
+	/**
+	 * Returns the product price to send to Alma's API to display the widget.
+	 *
+	 * @param WC_Product $product A WC product.
+	 * @return integer.
+	 */
+	private function get_price_to_inject_in_widget( $product ) {
+
+		if ( $product->is_type( 'simple' ) ) {
+			if ( version_compare( wc()->version, '3.0', '>=' ) ) {
+				$price = wc_get_price_including_tax( $product );
+			} else {
+				$price = $product->get_price_including_tax();
+			}
+		} elseif ( $product->is_type( 'variable' ) ) {
+			$price = $product->get_variation_regular_price( 'min', true );
+			if ( $product->is_on_sale() ) {
+				$price = $product->get_variation_sale_price( 'min', true );
+			}
+		}
```

Comment:
```suggestion
		if ( version_compare( wc()->version, '3.0', '>=' ) ) {
			$price = wc_get_price_including_tax( $product );
		} else {
			$price = $product->get_price_including_tax();
		}
		if ( $product->is_type( 'variable' ) && $product instanceof WC_Product_Variable ) {
			$price = $product->get_variation_regular_price( 'min', true );
			if ( $product->is_on_sale() ) {
				$price = $product->get_variation_sale_price( 'min', true );
			}
		}
```

Reply:
if you start with  `if ( $product->is_type( 'simple' ) ) {`,

`$price` is never initialized and should returns something else than an `integer`.

`get_variation_regular_price()` is only available in subclass `WC_Product_Variable`, so you have to verify that $product is instanceof of this class.

Reply:
`$price` is initialized on every case, so I don't understand your remark.

In my opinion `if ( $product->is_type( 'variable' )  )` returns true if $product is instance of `WC_Product_Variable`, so I think that you are trying to double check.

Reply:
ok for this.